src/lib/utils/paths.ts

```diff
@@ -0,0 +1,17 @@
+import * as Path from 'path';
+import { isArray } from 'util';
+import { Minimatch, IMinimatch } from 'minimatch';
+
+function createMinimatch(pattern: string): IMinimatch {
+  if (pattern[0] === '.' || pattern[0] === '/' || /^\w+(?!:)($|[/\\])/.test(pattern)) {
```

Comment:
Can you explain this regex? I don't have the skills to understand it.

Is there any of this that Minimatch handles already?

Reply:
Ok so this `if` is to check if a path like these:
- `./some/path`
- `../some/path`
- `some/path`
- `path`

Which are all relative paths and should be parsed relative to `process.cwd()` (`Path.resolve(path)`).

The regex is to test that windows absolute paths like `C:/some/path` wont be confused for a relative path like `c/some/path`. So I use a negative look ahead to ensure there is no `:` after the letters in the first part of the path (`path` will be a relative path and `path:` won't)

I could have used one regex to test for all of these cases, but it is a lot faster to do the simple tests first and only do the regex if really needed.

The main issue is that generally in other libraries that uses GLOB expressions you can just type `./some/path` and that would be treated as relative to the project root, but this have not been the case for this project where `**/some/path` was needed to ensure correct path matching, which has been counter-intuitive for many developers.

Reply:
But I can see I made a mistake: `pattern[0] === '/'` would match `/some/path` as relative, which isn't the case. I will fix this.